src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs

```diff
@@ -318,12 +318,46 @@ public IEnumerable<string> MakeCompliant(string name)
 
         private string CreateCompliantNameReusingPartialPrefixesAndSuffixes(string name)
         {
+            name = StripCommonPrefixes(name);
             name = EnsurePrefix(name);
             name = EnsureSuffix(name);
 
             return FinishFixingName(name);
         }
 
+        private static string StripCommonPrefixes(string name)
+        {
+            var index = 0;
+            while (name.Length > index + 1)
+            {
+                switch (char.ToLowerInvariant(name[index]))
```

Comment:
nit: I don't think the case indentation below matches the repo style.

Reply:
➡️ It doesn't. I tried to fix this in #22143 but the change was rejected. Until then the repo style is "user selection".

Reply:
:arrow_right: Fixed in b825107d439966101408335dd38645745c935f85